opentuner/search/technique.py

```diff
@@ -204,15 +223,20 @@ def call_main_generator(self):
                                     self.pending_tests)
         if self.pending_tests:
           self.rounds_since_novel_request = 0
-          yield None # wait
+          yield False # wait
 
 #list of all techniques
 the_registry = list()
 
+#list of composable techniques
+the_composable_registry = list()
 
 def register(t):
   the_registry.append(t)
 
+def register_composable(t):
```

Comment:
That's originally why I had a separate method, "register_composable". Using this method reflects an intent that "I want this composable technique instance to be considered when generating technique+operator combos", where using "register" means "I want this technique instance to be available for use as is." The first intent might be true without the second intent being true. If so we wouldn't want to "register" the composable technique instance.

Also, I don't think that just biasing probabilities of selection is any more elegant. Assume we have 2 composable classes A and B with instance a1, a2 ... and b1, b2 ... 
There are 2 cases.
1. Each class of a technique has it's own bias. We might have a1 and a2 with different hyper-parameters that we want to consider when generating, but we don't want to consider a3. Here, there is no way to tell that we don't want a3
2. Each instance of a technique has it's own bias. We might have a1, a2, a3, ... an, but we only want a1. The vast majority of instances should have weight=0.0 (which should probably be the default anyway). So if we want something considered, we have explicitly set a parameter (not too different from explicitly calling a separate register method). This case is better, but still doesn't handle if we don't want to use a1 as a search technique as is. 

That said, I think associating a weight with each registered composable technique is a great idea.